### PR TITLE
[DOC] ShapeInference.md update. slyalin comments

### PR DESCRIPTION
**Pull request by jane-intel**

### CODE REVIEW

File: docs/IE_DG/ShapeInference.md [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+   
+> **NOTES**:
+> - Starting with the 2021.1 release, the Model Optimizer converts topologies keeping shape-calculating sub-graphs by default, which enables correct shape propagation during reshaping in most cases.
+> - Older versions of IRs are not guaranteed to reshape successfully. Please regenerate them with the Model Optimizer of the latest version of OpenVINO™.<br>
+> - If an ONNX model does not have a fully defined input shape and the model was imported with the ONNX importer, reshape the model before loading it to the plugin.
+
+* **Set a new batch dimension value** with the `InferenceEngine::CNNNetwork::setBatchSize` method.<br>     
+   The meaning of a model batch may vary depending on the model design.
+   This method doesn't deduce batch placement for inputs of the model from the model architecture.
+   It assumes batch is placed under the index 0 in the shape for all the inputs and uses `InferenceEngine::CNNNetwork::reshape` method to propagate updated shapes through the model.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:17]:
```suggestion
   It assumes that the batch is placed at the 0 index in the shape for all inputs and uses the `InferenceEngine::CNNNetwork::reshape` method to propagate updated shapes through the model.
```

File: docs/IE_DG/ShapeInference.md [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+
+   The method transforms the model before a new shape propagation to relax a hard-coded batch dimension in the model, if any.
+
+   Use `InferenceEngine::CNNNetwork::reshape` instead of `InferenceEngine::CNNNetwork::setBatchSize` to set new input shapes for the model in case the model has:
+   * Multiple inputs with different 0 index dimension meaning
+   * Input without a batch dimension
+   * scalar, 1D, 3D input or input without a batch dimension
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:24]:
```suggestion
   * Multiple inputs with different 0 index dimension meanings
   * Input without a batch dimension
   * Scalar, 1D, or 3D input or input without a batch dimension
```

Doesn't the 3d point include the 2nd?

File: docs/IE_DG/ShapeInference.md [hunk 3]
```diff
@@ -1,6 +1,36 @@
...
+
+   `InferenceEngine::CNNNetwork::setBatchSize` method is a high level API method which wraps `InferenceEngine::CNNNetwork::reshape` method call and works for trivial models from the batch placement standpoint.
+   Please use `InferenceEngine::CNNNetwork::reshape` for all the other models.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:27]:
```suggestion
   The `InferenceEngine::CNNNetwork::setBatchSize` method is a high-level API method that wraps the `InferenceEngine::CNNNetwork::reshape` method call and works for trivial models from the batch placement standpoint.
   Use `InferenceEngine::CNNNetwork::reshape` for other models.
```

File: docs/IE_DG/ShapeInference.md [hunk 4]
```diff
@@ -1,6 +1,36 @@
...
+
+   Using `InferenceEngine::CNNNetwork::setBatchSize` method for models with a non-zero index batch placement or models with inputs without a batch dimension may lead to undefined behaviour.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:29]:
```suggestion
   Using the `InferenceEngine::CNNNetwork::setBatchSize` method for models with a non-zero index batch placement or for models with inputs that do not have a batch dimension may lead to undefined behaviour.
```

**Review by aalborov — CHANGES_REQUESTED** [review]

File: docs/IE_DG/ShapeInference.md [hunk 5]
```diff
@@ -1,6 +1,36 @@
...
+   The meaning of a model batch may vary depending on the model design.
+   This method does not deduce batch placement for inputs from the model architecture.
+   It assumes that the batch is placed at the zero index in the shape for all inputs and uses the `InferenceEngine::CNNNetwork::reshape` method to propagate updated shapes through the model.
+
+   The method transforms the model before a new shape propagation to relax a hard-coded batch dimension in the model, if any.
+
+   Use `InferenceEngine::CNNNetwork::reshape` instead of `InferenceEngine::CNNNetwork::setBatchSize` to set new input shapes for the model in case the model has:
+   * Multiple inputs with different zero index dimension meanings
+   * Input without a batch dimension
+   * Scalar, 1D, or 3D input or input without a batch dimension
```
Comment (jane-intel) [inline docs/IE_DG/ShapeInference.md:24]:
```suggestion
   * 0D, 1D, or 3D shape
```

File: docs/IE_DG/ShapeInference.md [hunk 6]
```diff
@@ -1,6 +1,36 @@
...
+
+* **Set a new batch dimension value** with the `InferenceEngine::CNNNetwork::setBatchSize` method.<br>     
+   The meaning of a model batch may vary depending on the model design.
+   This method does not deduce batch placement for inputs from the model architecture.
+   It assumes that the batch is placed at the zero index in the shape for all inputs and uses the `InferenceEngine::CNNNetwork::reshape` method to propagate updated shapes through the model.
+
+   The method transforms the model before a new shape propagation to relax a hard-coded batch dimension in the model, if any.
+
+   Use `InferenceEngine::CNNNetwork::reshape` instead of `InferenceEngine::CNNNetwork::setBatchSize` to set new input shapes for the model in case the model has:
+   * Multiple inputs with different zero index dimension meanings
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:22]:
```suggestion
   * Multiple inputs with different zero-index dimension meanings
```

**Review by aalborov — APPROVED** [review]
Just an afterthought on *zero-index dimension meanings*, please apply if it doesn't ruin the sense.

**Pull request merged**